plugins/ltac/tacinterp.ml

```diff
@@ -2017,6 +2017,31 @@ end
 (* globalization + interpretation *)
 
 
+let warning_fragile_intros =
+  CWarnings.create_with_quickfix ~default:CWarnings.Disabled ~name:"fragile-intros"
+    ~category:CWarnings.CoreCategories.pedantic
+    (fun () -> strbrk "intros is fragile, specify the names." ++ fnl ())
+
+let intro_patterns_deprecated b l loc : unit Proofview.tactic =
+  Proofview.Goal.enter begin fun gl ->
+    let old_hyps = Proofview.Goal.hyps gl in
+    Tactics.intro_patterns b l >>=
+    fun () -> Proofview.Goal.enter begin fun gl ->
+      let new_hyps = Proofview.Goal.hyps gl in
+      let old_ids = List.map Context.Named.Declaration.get_id old_hyps in
+      let new_ids = List.map Context.Named.Declaration.get_id new_hyps in
+      let nnew = List.length new_ids in
+      let nold = List.length old_ids in
+      let names, rest = CList.chop (nnew - nold) new_ids in
+      let names = List.rev names in
+      if names <> [] && Option.has_some loc then
+        warning_fragile_intros ?loc
```

Comment:
IIUC when the as-error flag is set, this would result in a backtracking error. I think this is wrong, it should be a fatal panic otherwise we're going to get extremely weird behaviours.

Reply:
Also, intros without names is not deprecated at all. Some use cases are perfectly legitimate, notably in proof automation.

Reply:
> Also, intros without names is not deprecated at all. Some use cases are perfectly legitimate, notably in proof automation.

My attempt is to capture just toplevel calls to intros, that is `intros.`

Reply:
> IIUC when the as-error flag is set, this would result in a backtracking error.

This is a problem, but I don't get that only affects this PR. Is there no tactic out there raising a warning?

Reply:
Moved to another PR, namely #19224